### ISSUE TITLE
MLJModelInterface 1.3.5 has broken MLJModels

### ISSUE THREAD

**Issue by ablaom** [JuliaAI/MLJModels.jl#423]
It seems MLJModelInterface 1.3.5 has broken MLJModels:

@OkonSamuel 

```julia
julia> X, y = @load_boston;

julia> models(matching(X, y))
WARNING: both ScientificTypes and MLJModelInterface export "scitype"; uses of it in module MLJModels must be qualified
ERROR: UndefVarError: scitype not defined
Stacktrace:
 [1] matching(X::NamedTuple{(:Crim, :Zn, :Indus, :NOx, :Rm, :Age, :Dis, :Rad, :Tax, :PTRatio, :Black, :LStat), NTuple{12, SubArray{Float64, 1, Matrix{Float64}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}}}, y::SubArray{Float64, 1, Matrix{Float64}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true})           
   @ MLJModels ~/.julia/packages/MLJModels/fsSbS/src/model_search.jl:198
 [2] top-level scope
   @ REPL[10]:1

```

**Comment by ablaom** [comment]
It seems this an issue for any package that simultaneously does `using MLJModelInterface` and `using ScientificTypes`. Note MLJModels cannot do `ScientificTypes -> ScientificTypesBase` because it needs the `coerce` function.

**Comment by ablaom** [comment]
I've made a fix for MLJModels at #424. 

However, on reflection, I think the lesser of two evils would be to be yank MLJModelInterface 1.3.5, MLJBase 0.19.2 and issue a patch for MLJFlux, the package broken by MLJModelInterface 1.3.4. The rationals are:

- Backwards compatability of MLJModels is more important that backwards compatibility of MLJFlux
- MLJModelInterface 1.3.4 -> 1.3.5 is really a backwards step in the general plan of exporting *less* from MLJModelInterface (https://github.com/JuliaAI/MLJModelInterface.jl/issues/132). 

@OkonSamuel Thoughts?

**Comment by OkonSamuel** [comment]
> I've made a fix for MLJModels at #424. You can see this is a bit awkward.
> 
> However, on reflection, I think the lesser of two evils would be to be yank MLJModelInterface 1.3.5, MLJBase 0.19.2 and issue a patch for MLJFlux, the package broken by MLJModelInterface 1.3.4. The rationals are:
> 
> * Backwards compatability of MLJModels is more important that backwards compatibility of MLJFlux
> * MLJModelInterface 1.3.4 -> 1.3.5 is really a backwards step in the general plan of exporting _less_ from MLJModelInterface ([Perhaps MLJModelInteface should not export any method in the future MLJModelInterface.jl#132](https://github.com/JuliaAI/MLJModelInterface.jl/issues/132)).
> 
> @OkonSamuel Thoughts?

I agree with you on this. We shouldn't export the `scitype` method from MLJModelInterface. But we should still leave the newly added `scitype` method there.

**Comment by ablaom** [comment]
> But we should still leave the newly added scitype method there.

👍🏾

**Comment by ablaom** [comment]
Actually, backwards compatibility of MLJModels is less of an issue as we just tagged a breaking version, which can be yanked. So, after further investigation and off-line discussion @OkonSamuel  and I are recommending merging the fix #424 after all.
